### PR TITLE
Add new REopt API variables, debug

### PR DESCRIPTION
**Pull request by lixiangk1**
Added new output variables for REopt and debugged fractional GHX sizing code.

@Bill-Becker The toml files were updated a lot with packages I used to do some local debugging, not sure if or how I should revert them?

### CODE REVIEW

**Comment by Bill-Becker** [comment]
Any packages used just for testing or debugging should just stay local or if you need packages for unit tests that are added, they should be in the /test .toml files. So please revert the package additions to the main environment if they are not required for the package.

File: Project.toml [hunk 1]
```diff
@@ -3,5 +3,11 @@ uuid = "7ce85f02-24a8-4d69-a3f0-14b5daa7d30c"
...
+[deps]
+CSV = "336ed68f-0bac-5ca0-87d4-7b16caf5d00b"
+DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
+HTTP = "cd3eb016-35fb-5094-929b-558a96fad6f3"
+JSON = "682c06a0-de6a-54ab-a142-c8b1cf79cde6"
+
 [compat]
-julia = "1.7"
\ No newline at end of file
+julia = "1.7"
```
Comment (Bill-Becker) [inline Project.toml]: You can still update the package environment to julia 1.8, and then please add/append 1.8 to the julia compat versions, i.e. julia = "1.7, 1.8"

**Review by Bill-Becker — APPROVED** [review]
I'm approving but please remove those packages that were added to the main environment files before merging. Thanks!

File: Project.toml (hunk 1 above)
Comment (lixiangk1) [reply Project.toml]: @Bill-Becker I removed the packages from the main environment. Just wanted to get your eyes on it one last time before I merged - wasn't too sure what `project_hash` is and wanted to make sure I'm not updating anything unexpectedly!

**Review by Bill-Becker — APPROVED** [review]

**Pull request merged**